tests/acceptance/test_update.py

```diff
@@ -78,6 +82,35 @@ def get_env_checksums(bitbake_variables):
 
         return checksums
 
+    @staticmethod
+    def corrupt_middle_byte(fd):
+        # Corrupt the middle byte in the contents.
+        middle = int(os.fstat(fd.fileno()).st_size / 2)
+        fd.seek(middle)
+        middle_byte = int(fd.read(1).encode("hex"), base=16)
+        fd.seek(middle)
+        # Flip lowest bit.
+        fd.write("%c" % (middle_byte ^ 0x1))
```

Comment:
Just a heads up here. The signature file is base64 encoded data you are reading from file and then you are flipping some "random" bit. Are you sure we won't end up with invalid base64 data at the end?